File: app/src/main/java/com/example/supergrocery/ROOM/ItemsDao.java
package com.example.supergrocery.ROOM;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ItemsDao {

    @Insert
    void insert(OrderItem items);

    @Update
    void update(OrderItem items);

    @Query("DELETE FROM OrderItems")
    void deleteAll();

    @Query("SELECT * FROM OrderItems ORDER BY id")
    List<OrderItem> getAllItems();

    @Delete
    void delete(OrderItem itemModel);

}

File: app/src/main/java/com/example/supergrocery/room/ItemsDao.kt
package com.example.supergrocery.room

import androidx.room.*

@Dao
interface ItemsDao {
    @Insert
    suspend fun insert(items: OrderItem)

    @Update
    suspend fun update(items: OrderItem)

    @Query("DELETE FROM OrderItems")
    suspend fun deleteAll()

    @Query("SELECT * FROM OrderItems ORDER BY id")
    suspend fun getAllItems(): List<OrderItem>

    @Delete
    suspend fun delete(itemModel: OrderItem)

    @Query("UPDATE OrderItems SET quantity = quantity + 1 WHERE id = :id")
    suspend fun incrementQuantity(id: Int)

    @Query("UPDATE OrderItems SET quantity = quantity - 1 WHERE id = :id")
    suspend fun decrementQuantity(id: Int)

    @Query("SELECT COUNT(id) FROM OrderItems WHERE id = :id")
    suspend fun checkInBasket(id: Int): Int

    @Query("SELECT COUNT(quantity) FROM OrderItems WHERE id = :id")
    suspend fun checkQuantityBasket(id: Int): Int
}